### DISCUSSION TITLE
Predicting an edge type which isn't present in test graphs

### DISCUSSION THREAD

**Discussion by DoctorDinosaur** [pyg-team/pytorch_geometric#9573] [Q&A]
I have three edge types and one node type, across a dataset of train/valiation/test graphs with different numbers of nodes. I want to predict all occurances of `type3` on unseen test graphs, which initially have no `type3` edges.

From `examples/mutag_gin.py` it seems the training on multiple graphs, and testing on unseen, is simple. I just apply the trained model.

But Message Passing and Negative Sampling is confusing me for my case where `type3` edges aren't present in the test set, i.e. I can't use `type3` for message passing. And I'm not sure of a good model structure.

https://github.com/pyg-team/pytorch_geometric/discussions/8899 goes into some issues, but it's not fully resolved.

Are there any examples which would be useful to me, here? I'm new to GNNs, including the more basic concepts.
I'd like to see examples of, or reading materials for:
- Setting up the train/validation set to exclude `type3` from message passing, since it won't be present on test graphs. Is it that I just put i.e. GATv2conv layers between `type1` and `type2`?
- Setting up negative sampling for this heterogenous link prediction
- An example of the `predict` and `forward` step for this case
- A recommended model structure. Would it be something like a linear combination of GAT layers, followed by fully connected linear layers? What should the GAT convolution be between?
